lldb/source/Plugins/ExpressionParser/Clang/ClangExpressionParser.cpp

```diff
@@ -471,6 +498,18 @@ static void SetupTargetOpts(CompilerInstance &compiler,
   // Set the target ABI
   if (std::string abi = GetClangTargetABI(target_arch); !abi.empty())
     compiler.getTargetOpts().ABI = std::move(abi);
+
+  if ((target_machine == llvm::Triple::riscv64 &&
+       compiler.getTargetOpts().ABI == "lp64f") ||
+      (target_machine == llvm::Triple::riscv32 &&
+       compiler.getTargetOpts().ABI == "ilp32f"))
+    compiler->getTargetOpts().FeaturesAsWritten.emplace_back("+f");
+
+  if ((target_machine == llvm::Triple::riscv64 &&
+       compiler.getTargetOpts().ABI == "lp64d") ||
+      (target_machine == llvm::Triple::riscv32 &&
+       compiler.getTargetOpts().ABI == "ilp32d"))
+    compiler->getTargetOpts().FeaturesAsWritten.emplace_back("+d");
```

Comment:
What happens if we just unconditionally add `+d` and `+f`? I'm not super familiar with RISCV but are ilp32d/ilp32f/ilp64f/ilp32f all the supported ABI strings? The GCC docs also seem to mention `rv64ifd`, e.g.. So my thinking is that we could maybe just enable these features for all targets (like we do with x86 sse?)

Reply:
Makes sense, thanks!